src/environ/secrets.py

```diff
@@ -113,13 +135,80 @@ def _get(self, environ, metadata, prefix, name):
             var = "_".join((prefix[1:] + (name,)))
         try:
             log.debug("looking for '%s' in section '%s'." % (var, section))
-            return _SecretStr(self._cfg.get(section, var))
+            val = self._cfg.get(section, var)
+            return _SecretStr(val)
         except NoOptionError:
-            if isinstance(ce.default, attr.Factory):
-                return attr.NOTHING
-            elif not isinstance(ce.default, Raise):
-                return ce.default
-            raise MissingSecretError(var)
+            return _get_default_secret(var, ce.default)
+
+
+@attr.s
+class DirectorySecrets(object):
+    """Load secrets from a directory containing secrets in separate files.
```

Comment:
```suggestion
    """
    Load secrets from a directory containing secrets in separate files.
```

Reply:
A lot of tools (Sphinx too) parse the docstring first line as summary, and the rest is the details. In styling, this is usually looks different. I would leave this.

Reply:
That's true, but then this would be the only docstring with that style. :) I follow the same style that Django and Twisted do because it's prettier and more readable in source.